modules/core/src/shaderlib/shadow/shadow.ts

```diff
@@ -206,34 +206,34 @@ function getViewProjectionMatrices({
 
 // eslint-disable-next-line complexity
 function createShadowUniforms(
-  opts: ShadowModuleProps
+  opts: Partial<ShadowModuleProps>
 ): ShadowModuleBindings & ShadowModuleUniforms {
   const {shadowEnabled = true} = opts;
   if (!shadowEnabled || !opts.shadowMatrices || !opts.shadowMatrices.length) {
     return {
       drawShadowMap: false,
       useShadowMap: false,
-      shadow_uShadowMap0: opts.dummyShadowMap,
-      shadow_uShadowMap1: opts.dummyShadowMap
+      shadow_uShadowMap0: opts.dummyShadowMap!,
+      shadow_uShadowMap1: opts.dummyShadowMap!
     };
   }
   const projectUniforms = project.getUniforms(opts) as ProjectUniforms;
```

Comment:
@ibgreen In v8 `getUniforms` can access uniforms created by this shader module's dependencies (`project` in this case). As of the latest master, several shader modules are calling `project.getUniforms` themselves, which technically is incorrect, because they do not receive the full set of project module props (modelMatrix, coordinateOrigin, etc.). Is this by design, or is it a regression?

Reply:
- I wasn't really aware of this usage, so I suppose it was at most semi-intentional, the goal was simply to have a clean, simple-to-understand API.
- If it is a crucial feature we can consider adding it back, but I fear that making types of dependency modules flow into a dependent module could require more typescript machinery than it is worth.
- Perhaps we can look at the cases where it is used and see what are options are.

Reply:
Related: https://github.com/visgl/luma.gl/issues/2138

My view is that it is too much magic to have the dependent module uniforms be available, and not as flexible. I started adding the `project.getUniforms` pattern as it feels more clear what is going on and all the uniform generation logic is then happening in one place and at the same time. I don't think it is a problem to pass the project module props where needed